app/dao/services_dao.py

```diff
@@ -194,16 +197,38 @@ def dao_fetch_service_by_inbound_number(number):
 
 
 def dao_fetch_service_by_id_with_api_keys(service_id, only_active=False):
-    query = Service.query.filter_by(
-        id=service_id
-    ).options(
-        joinedload('api_keys')
-    )
-
-    if only_active:
-        query = query.filter(Service.active)
+    @contextmanager
+    def get_session(engine):
+        session = scoped_session(sessionmaker(bind=engine))
+        try:
+            yield session
+        finally:
+            session.close()
```

Comment:
This is going to be used in a lot of places once we start using the read db more widely, right?  Perhaps it should be moved to app/utils.py and imported.